plugins/4.mediaPlayer.ts

```diff
@@ -12,8 +13,10 @@ export default defineNuxtPlugin((_) => {
   return {
     provide: {
       mediaPlayer: initMediaPlayer(
-        (src) => new Audio(src),
-        getAccessTokenSilently,
+        (src) =>
+          new MediaTrack(() =>
+            getAccessTokenSilently().then((t) => authorizedUrl(src, t)),
```

Comment:
The understanding from Andreas was that getAccessTokenSilently() is cached and that we don't need to optimize that.
Did you verify that that is not the case? Or is that an assumption (which might be wrong)?

Reply:
I haven't changed anything on the workflow here. The reference has just been moved out of the class `MediaPlayer`.

The assumption, that `getAccessTokenSilently()` caches the token and only requests a fresh token when expired is true as long as `cacheMode` is set to `on`, which is the default - https://auth0.github.io/auth0-vue/interfaces/GetTokenSilentlyOptions.html#cacheMode.